src/error.rs

```diff
@@ -57,6 +57,18 @@ impl From<miniscript::Error> for ScriptCreationError {
     }
 }
 
+impl PartialEq for ScriptCreationError {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::PolicyCompilation(l0), Self::PolicyCompilation(r0)) => l0 == r0,
+            (Self::MiniscriptError(l0), Self::MiniscriptError(r0)) => {
+                l0.to_string() == r0.to_string()
+            }
+            _ => core::mem::discriminant(self) == core::mem::discriminant(other),
```

Comment:
Might be a good idea to newtype `miniscript::Error` and `miniscript::policy::compiler`. Then manual `PartialEq` implementations would only be necessary on those newtypes, and `derive` can handle the rest.

As it stands, adding a new data enum field would necessitate updating the manual implementation added in this PR, because `core::mem::discriminant` disregards the data carried by enums.

Reply:
I'm not sure if we should implement `PartialEq` ourselves. The use of `core::mem::discriminant` looks like a nice footgun, and we are very close from being able to directly derive `PartialEq` (see https://github.com/rust-bitcoin/rust-miniscript/pull/358) now that some background work  was done to derive PartialEq in most rust-bitcoin types in previous releases.

If this is only to keep the one check touched by this PR (https://github.com/revault/revault_tx/pull/123/files#diff-b588eb28cc6dc24d79bbc2ed58b01cae448429f9b173910edebbad59332c6fbdR1631-R1632), i think just comparing the `to_string()` of the errors is a cleaner hack to have before we can cleanly have `PartialEq`.

Reply:
What do you think of:
```rust
impl PartialEq for ScriptCreationError {
    fn eq(&self, other: &Self) -> bool {
        self.to_string() == other.to_string()
    }
}
```
So that when the time comes it can be removed and no need for `.to_string()` across the code?

Reply:
Sounds ok to me. Please leave a `TODO` along with the PartialEq implem :)